spec/models/user_option_spec.rb

```diff
@@ -18,6 +18,28 @@
 
   end
 
+  describe ".mailing_list_mode" do
+    let!(:forum_user) { Fabricate(:user) }
+    let!(:mailing_list_user) { Fabricate(:user) }
+
+    before do
+      forum_user.user_option.update(mailing_list_mode: false)
+      mailing_list_user.user_option.update(mailing_list_mode: true)
+    end
+
+    it "should return false when `SiteSetting.disable_mailing_list_mode` is enabled" do
+      SiteSetting.expects(:disable_mailing_list_mode).twice.returns(true)
```

Comment:
I would try to avoid mocking the site settings here since we can easily set it and ensure we revert to the default in the test.

Any thoughts on this? @ZogStriP :)